soccer/src/soccer/server/ball_placement_server.cpp

```diff
@@ -0,0 +1,63 @@
+#include "ball_placement_server.hpp"
+
+namespace server {
+using BallPlacement = rj_msgs::action::BallPlacement;
+using GoalHandleBallPlacement = rclcpp_action::ServerGoalHandle<BallPlacement>;
+BallPlacementActionServer ::BallPlacementActionServer(const rclcpp::NodeOptions& options)
+    : Node("ball_placement_server", options) {
+    using namespace std::placeholders;
+
+    this->action_server_ = rclcpp_action::create_server<BallPlacement>(
+        this->get_node_base_interface(), this->get_node_clock_interface(),
+        this->get_node_logging_interface(), this->get_node_waitables_interface(), "ball_placement",
+        std::bind(&BallPlacementActionServer::handle_goal, this, _1, _2),
+        std::bind(&BallPlacementActionServer::handle_cancel, this, _1),
+        std::bind(&BallPlacementActionServer::handle_accepted, this, _1));
+}
+
+rclcpp_action::GoalResponse BallPlacementActionServer ::handle_goal(
+    const rclcpp_action::GoalUUID& uuid, std::shared_ptr<const BallPlacement::Goal> goal) {
+    (void)uuid;
+
+    return rclcpp_action::GoalResponse::ACCEPT_AND_EXECUTE;
+}
+
+rclcpp_action::CancelResponse BallPlacementActionServer ::handle_cancel(
+    const std::shared_ptr<GoalHandleBallPlacement> goal_handle) {
+    (void)goal_handle;
+    return rclcpp_action::CancelResponse::ACCEPT;
+}
+
+void BallPlacementActionServer ::handle_accepted(
+    const std::shared_ptr<GoalHandleBallPlacement> goal_handle) {
+    // this needs to return quickly to avoid blocking the executor, so spin up a new thread
+    using namespace std::placeholders;
+    std::thread{std::bind(&BallPlacementActionServer::execute, this, _1), goal_handle}.detach();
+}
+
+void BallPlacementActionServer ::execute(
+    const std::shared_ptr<GoalHandleBallPlacement> goal_handle) {
+    std::shared_ptr<const BallPlacement::Goal> goal = goal_handle->get_goal();
+    rj_msgs::msg::ServerIntent server_intent = goal->server_intent;
+    rj_msgs::msg::RobotIntent robot_intent = server_intent.intent;
+    int robot_id = server_intent.robot_id;
```

Comment:
so for the execute loop, I think what needs to happen is:
 * find vector from current ball pt to goal
 * call path target motion planner to move behind that vector
 * call collect planner to get ball in dribbler
 * write custom planner to move at a slow rate while dribbler is on (not entirely sure about this part, should be an adaptation of either path-target or collect planner though) 
 * OR call path-target to a different robot to position its mouth at goal, and call line kick planner (or kick directly since in theory we are facing the right way?)

The collect planner tries to calculate which direction it should approach the ball based on the ball's current velocity, so for ball placement it would be better just to force it on the right line (since goal is known). We can probably shove all this logic into the ball placement server for now and separate it into a new class if needed later

Reply:
we should just be able to publish a trajectory directly from this node, that's what planner node does: https://github.com/RoboJackets/robocup-software/blob/ad8cb935bfc58a84337fe02be046234c4c9b368e/soccer/src/soccer/planning/planner_node.cpp#L49-L51

Reply:
if we decided to run ball placement through the planner node it might be more painful, since we would have to make a new planner that does all the steps above and then pub a RobotIntent to that new planner. it could be done, though